wave_vision/CMakeLists.txt

```diff
@@ -32,7 +17,19 @@ ADD_LIBRARY(
     src/matcher/brute_force_matcher.cpp
     src/matcher/flann_matcher.cpp
 )
-TARGET_LINK_LIBRARIES(wave_vision ${WAVE_VISION_DEPS})
+
+# Use these headers when building, and make clients use them
+TARGET_INCLUDE_DIRECTORIES(${PROJECT_NAME} PUBLIC include)
+
+# Depend on these modules and external libraries, and make clients use them
+TARGET_LINK_LIBRARIES(${PROJECT_NAME} PUBLIC
+    wave_kinematics
+    wave_utils
+    wave_containers
+    Eigen3::Eigen
+    Boost::filesystem
+    ${OpenCV_LIBS})
```

Comment:
is it ok to keep this as OpenCV_libs? Should this be updated to reflect the components that are linked? or is too many components?

Reply:
`OpenCV_libs` only includes the requested components

Reply:
cool. But you changed it above in utils to just include opencv_core?

Reply:
Yes, utils only uses opencv_core.
In next PR I move all find_package to the top level anyway